src/Meshes.jl

```diff
@@ -305,6 +305,7 @@ export
   OverlappingBoxes,
   FaceTouchingBoxes,
   CornerTouchingBoxes,
+  RayCrossingBox,
```

Comment:
Can we rename this to `IntersectingRayBox` for consistency with other names?

Reply:
In fact, there may be another condition. `RayTouchingBox` contains a `Point` instead of a `Segment`. Use this name to reserve space for that possibility.

You may ask why have not write that condition. Because that is not stable catches due to floating point error. But maybe we got a great idea in the future?

Reply:
Nice, knowing that there are other cases not covered yet, it now makes more sense to choose a more specific name. What about `CrossingRayBox`? We place the verb first to match the other cases with segments an boxes.

Regarding the floating point issue, I was thinking we could at least leave the branch in the algorithm if that is possible. In the future we can come back with interval arithmetic to fix it.